kubeconfig_plugin.py

```diff
@@ -111,63 +111,151 @@ class ErrorOutput:
     id="kubeconfig",
     name="kubeconfig plugin",
     description=(
-        "Inputs a kubeconfig, parses it and extracts the kubernetes cluster"
-        " details "
+        "Inputs a kubeconfig, parses it and extracts the kubernetes cluster details"
     ),
     outputs={"success": SuccessOutput, "error": ErrorOutput},
 )
 def extract_kubeconfig(
     params: InputParams,
 ) -> typing.Tuple[str, typing.Union[SuccessOutput, ErrorOutput]]:
-
     print("==>> Parsing and extracting kubernetes cluster details ...")
 
     try:
-        kubeconfig = yaml.safe_load(params.kubeconfig)
-        kcl = config.kube_config.KubeConfigLoader(kubeconfig)
+        try:
+            kubeconfig = yaml.safe_load(params.kubeconfig)
+        except Exception as e:
+            return "error", ErrorOutput(
+                "Exception occurred while loading YAML. Input is not valid YAML."
+                + " Exception: {}".format(e.__str__())
+            )
 
-        output = {}
-        output["cluster_name"] = kcl._current_context["context"]["cluster"]
-        output["server_url"] = kcl._cluster.value["server"]
-        output["certificate_authority_data"] = kcl._cluster.value[
-            "certificate-authority-data"
-        ]
-        output["user"] = kcl._current_context["context"]["user"]
+        # Kubeconfig files have the kind set as Config
+        if "kind" in kubeconfig:
+            if kubeconfig["kind"] != "Config":
+                return "error", ErrorOutput(
+                    "The provided file is not a kubeconfig file"
+                )
+        else:
+            return "error", ErrorOutput(
+                "The provided file is not a kubeconfig file (missing 'kind' field)"
+            )
```

Comment:
Consider a couple of alternatives if you are gonna be back in here:

```
        if "kind" not in kubeconfig:
            return "error", ErrorOutput(
                "The provided file is not a kubeconfig file (missing 'kind' field)"
            )
        if kubeconfig["kind"] != "Config":
            return "error", ErrorOutput(
                "The provided file is not a kubeconfig file"
            )
```

or a bit longer but you leverage a more pythonic way to do this where you only reference the dictionary once:

```
        try:
            kind = kubeconfig["kind"]
        except KeyError:
            return "error", ErrorOutput(
                "The provided file is not a kubeconfig file (missing 'kind' field)"
            )
        if kind != "Config":
            return "error", ErrorOutput(
                "The provided file is not a kubeconfig file"
            )
```

Reply:
And this notion of checking a dictionary and then accessing the dictionary for the same element is littered below.  It is usually worth the effort to only access the dictionary once.